syntax = "proto3";
package voxa.speech.v1;

option go_package = "github.com/jmarc101/voxa/api/gen/voxa/speech/v1;speechv1";

// AudioFrame carries a single ordered chunk of audio bytes.
message AudioFrame {
  // The encoding of the audio data (e.g., PCM16 or OPUS).
  AudioFormat format = 1;
  // The audio bytes. For PCM16, little-endian, mono. For OPUS, a single Opus packet.
  bytes data = 2;
  // A monotonically increasing sequence number for this stream (starts at 1).
  int64 sequence = 3;
}

// AudioFormat specifies the encoding of the synthesized audio.
enum AudioFormat {
  // The audio format is unspecified.
  AUDIO_FORMAT_UNSPECIFIED = 0;
  // 16-bit signed little-endian PCM audio, mono channel.
  AUDIO_FORMAT_PCM16 = 1;
  // Opus-encoded audio, mono channel.
  AUDIO_FORMAT_OPUS = 2;
}
